src/contexts/AuthContext.tsx

```diff
@@ -19,12 +20,24 @@ export function AuthProvider({ children }: IProps) {
       .then((cred) => {
         if (cred.user) {
           cred.user.updateProfile({ displayName: name });
+          signupFirestore(cred.user.uid, cred.user.email, name);
         }
       });
   }
 
   function login(email: string, password: string) {
-    return auth.signInWithEmailAndPassword(email, password);
+    return auth.signInWithEmailAndPassword(email, password).then((cred) => {
+      if (cred.user) {
+        // Backfill user displayName from Firestore to Auth
+        // only if the displayName doesn't already exist in Auth.
+        const name = getNameFirestore(cred.user.uid);
+        if (name && !cred.user.displayName) {
+          cred.user.updateProfile({ displayName: name });
+        }
+        // Backfill missing data from Auth to Firestore.
+        signupFirestore(cred.user.uid, cred.user.email, cred.user.displayName);
+      }
+    });
```

Comment:
If you take my suggestion for `getNameFirestore`, this could become:

```suggestion
    return auth.signInWithEmailAndPassword(email, password).then(async (cred) => {
      if (cred.user) {
        // Backfill user displayName from Firestore to Auth
        // only if the displayName doesn't already exist in Auth.
        const name = await getNameFirestore(cred.user.uid);
        if (name && !cred.user.displayName) {
          cred.user.updateProfile({ displayName: name });
        }
        // Backfill missing data from Auth to Firestore.
        signupFirestore(cred.user.uid, cred.user.email, cred.user.displayName);
      }
    });
```

Or you could do a `.then` off of `getNameFirestore` if you prefer that.

Reply:
Oh, since we're [already using login like it's async](https://github.com/anyatokar/mfte-seattle/blob/5688278827778bf7c1bd5d4dacd531f0eaacbc36/src/auth_components/Login.tsx#L32), you could convert this whole function to `async`.

Reply:
thanks! updated here 0b8c99c